### ISSUE TITLE
Could not load type 'Microsoft.AspNetCore.Razor.Language.RazorTemplateEngine' 

### ISSUE THREAD

**Issue by cheekean82** [TanvirArjel/TanvirArjel.Extensions.Microsoft.DependencyInjection#1]
Hi,

I'm using visual studio community. Whenever my PackageReference include

Microsoft.VisualStudio.Web.CodeGeneration.Design  Version="3.1.2"

I will have an error below. Do you have any idea how to solve this? Full error message at below

ReflectionTypeLoadException: Unable to load one or more of the requested types.
Could not load type 'Microsoft.AspNetCore.Razor.Language.RazorTemplateEngine' from assembly 'Microsoft.AspNetCore.Razor.Language, Version=3.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.

**Comment by TanvirArjel** [comment]
@cheekean82 Can you share a sample project so that I can reproduce the issue.?

**Comment by cheekean82** [comment]
Below is the sample project that I had created. I notice that another Nuget "captcha" causing this problem. 

https://github.com/cheekean82/test

You can see on the csproj file, there is 3 package reference. if enable either 2 it is working. if 3 together it will show the error.

dynamic + captcha = work
captcha + design = work
dynamic + design = work
dynamic + captcha + design = throw error

**Comment by TanvirArjel** [comment]
@cheekean82 Thank you so much for the details information. I shall take a deep into it and keep you updated.

**Comment by TanvirArjel** [comment]
@cheekean82 After lots of investigation I have come to the conclusion that this issue is not anyway related to `AspNetCore.ServiceRegistration.Dynamic` library. This is maybe happening due to some assembly version mismatch between the `Captcha` and  `Microsoft.VisualStudio.Web.CodeGeneration.Design`.

**Comment by TanvirArjel** [comment]
@cheekean82 By the way, I am adding a workaround to the library to overcome this type of situation. It will be available on version `1.2.0`

**Comment by cheekean82** [comment]
@TanvirArjel Thanks for the hard work you had been done. keep up the good work!!!! stay safe...

**Comment by TanvirArjel** [comment]
@cheekean82 Hi, I have released version `1.2.0` which contains a fix for your problem. To fix your problem do as follows:

```
public static void ConfigureServices(IServiceCollection services)
{
    // Assemblies start with "Microsoft.AspNetCore", "Microsoft.Extensions" will be ignored during type scanning.
    string[] assembliesToBeIngored = new string[] { "Microsoft.AspNetCore", "Microsoft.Extensions" };
    services.AddServicesOfType<IScopedService>(assembliesToBeIngored);
    services.AddServicesWithAttributeOfType<ScopedServiceAttribute>(assembliesToBeIngored);
}
```
